File: frontend/lib/models/address.dart
class Address {
  final String id;
  final String addressLine;
  final String? phoneNumber;
  final String? addressCategory;
  final String? addressTag;

  const Address({
    required this.id,
    required this.addressLine,
    this.phoneNumber,
    this.addressCategory,
    this.addressTag,
  });

  factory Address.fromJson(Map<String, dynamic> json) => Address(
        id: json['id'] as String,
        addressLine: json['addressLine'] as String,
        phoneNumber: json['phoneNumber'] as String?,
        addressCategory: json['addressCategory'] as String?,
        addressTag: json['addressTag'] as String?,
      );

  String get displayTitle => addressTag?.isNotEmpty == true ? addressTag! : addressCategory ?? 'Address';
}

File: order-together-backend/src/main/java/com/ordertogether/model/Address.java
package com.ordertogether.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

/**
 * A saved delivery address from the user's Swiggy account, as returned by the
 * {@code get_addresses} MCP tool.
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record Address(
		String id,
		String addressLine,
		String phoneNumber,
		String addressCategory,
		String addressTag) {
}
